### PR TITLE
Ensure dest rate of 0 does not throw for settlement (part of SIP-86)

### PR DESCRIPTION
**Pull request by justinjmoses**

### CODE REVIEW

File: contracts/ExchangeRates.sol [hunk 1]
```diff
@@ -272,6 +272,11 @@ contract ExchangeRates is Owned, SelfDestructible, MixinResolver, MixinSystemSet
 
         (uint srcRate, ) = _getRateAndTimestampAtRound(sourceCurrencyKey, roundIdForSrc);
         (uint destRate, ) = _getRateAndTimestampAtRound(destinationCurrencyKey, roundIdForDest);
+        if (destRate == 0) {
```
Comment (jacko125) [inline contracts/ExchangeRates.sol:275]: good catch on this division edge case !

**Review by zyzek — APPROVED** [review]

File: contracts/ExchangeRates.sol (hunk 1 above)
Comment (jacko125) [reply contracts/ExchangeRates.sol:275]: And will revert if `destRate` is 0 for `effectiveValueAndRates`

https://github.com/Synthetixio/synthetix/blob/master/contracts/ExchangeRates.sol#L607

**Review by jacko125 — DISMISSED** [review]

File: contracts/ExchangeRates.sol (hunk 1 above)
Comment (justinjmoses) [reply contracts/ExchangeRates.sol:275]: Hmmm. Should we prevent the revert in that case? Feels like it is the right move.

Comment (justinjmoses) [reply contracts/ExchangeRates.sol:280]: Done.

Comment (jacko125) [reply contracts/ExchangeRates.sol:280]: @justinjmoses  what was done ?

Comment (justinjmoses) [reply contracts/ExchangeRates.sol:280]: Adds the fix to `effectiveValueAndRates`: https://github.com/Synthetixio/synthetix/pull/699/files#diff-3477eb31675538d20c3531997ee9d0d6R623-R625

**Review by clementbalestrat — APPROVED** [review]

**Review by jacko125 — APPROVED** [review]

**Pull request merged**